File: Entities/SchoolDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TournamentManagementAdmin.Entities
{
    public partial class SchoolDetail
    {
        [Column("SchoolDetailID")]
        public int SchoolDetailId { get; set; }
        public DateTime CreateDate { get; set; }
        [Column("DivisionID")]
        public int? DivisionId { get; set; }
        [Column("LevelID")]
        public int? LevelId { get; set; }
        [Column("SchoolID")]
        public int? SchoolId { get; set; }
        [Column("SectorID")]
        public int? SectorId { get; set; }

        [ForeignKey("DivisionId")]
        [InverseProperty("SchoolDetail")]
        public Division Division { get; set; }
        [ForeignKey("LevelId")]
        [InverseProperty("SchoolDetail")]
        public Level Level { get; set; }
        [ForeignKey("SchoolId")]
        [InverseProperty("SchoolDetail")]
        public School School { get; set; }
        [ForeignKey("SectorId")]
        [InverseProperty("SchoolDetail")]
        public Sector Sector { get; set; }
    }
}

File: ClientApp/app/models/SchoolDetail.ts
import { Division } from "./Division";
import { Level } from "./Level";
import { School } from "./School";
import { Sector } from "./Sector";


export class SchoolDetail {
	constructor(
		schoolDetailId?: number,
		createDate?: Date,
		divisionId?: number,
		levelId?: number,
		schoolId?: number,
		sectorId?: number,
		division?: Division,
		level?: Level,
		school?: School,
		sector?: Sector) { };
	}
